server/src/main/java/org/opensearch/snapshots/SnapshotDeletionException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.snapshots;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+
+/**
+ * Thrown if requested snapshot/s can't be deleted
+ *
+ * @opensearch.internal
+ */
+public class SnapshotDeletionException extends SnapshotException {
```

Comment:
I think this is too general of a name for something that represents a snapshot in use. Rename to something more specific like `SnapshotHasIndicesDeletionException` or `SnapshotInUseDeletionException`? Also do we want to include the list of indices in the exception?

Reply:
SnapshotInUseDeletionException sounds better to me than the generic one. We are failing the complete request even when it has 1 snapshot that is not backing any index, so having the indices list in the error message would be helpful to let the user know exactly why we are failing the request. For eg, let's say snapshots 1 and 2 are remote snapshots backing an index and snapshot 3 doesn't back anything. When a delete request comes in the form of delete(1,2,3), we let the user know that 1,2 are remote snapshots backing an index and hence can't be deleted.

Reply:
I would double check that we're not leaking any information we shouldn't (names of indices), and otherwise +1.